src/prove.rs

```diff
@@ -7,9 +7,9 @@
 ///   more[...]
 ///   more2[...]
 /// ```
-pub trait TryProve<'a, T> {
-    type Error;
+pub trait TryProve<T> {
     type Proven;
+    type Error;
 
-    fn try_prove(&'a self, candidate: &'a T) -> Result<&'a Self::Proven, Self::Error>;
+    fn try_prove(self, candidate: T) -> Result<Self::Proven, Self::Error>;
 }
```

Comment:
Yeah - this reminds me I meant to comment on that earlier: It's totally valid to make these things references, but I think it's surprising to attach the reference lifetime to the struct itself. I.e. the previous version I'd have expected to look like this:
```rs
pub trait TryProve<T> {
    type Proven;
    type Error;

    fn try_prove<'a>(&'a self, candidate: &'a T) -> Result<&'a Self::Proven, Self::Error>;
}
```

It's just ever so slightly different than a new version where you'd `impl TryProve<&'a MyStruct, &'a MyStruct, Proven = &'a MyStruct>`.
Just noting here as another possibility.

Reply:
I just pushed my work from the past few days, and this entire trait no longer exists 😅